packages/time-picker/src/vaadin-time-picker.js

```diff
@@ -599,6 +599,12 @@ class TimePicker extends PatternMixin(InputControlMixin(ThemableMixin(ElementMix
     const parsedObj = this.__parseISO(value);
     const newValue = this.__formatISO(parsedObj) || '';
 
+    // Mark value set programmatically by the user
+    // as committed for the change event detection.
+    if (!this.__skipCommittedValueUpdate) {
```

Comment:
It's important to execute this part before `__updateInputValue()`. When `step` >= 1, `__updateInputValue()` can trigger the value observer again while the first call is still executing.